src/LinuxTracing/ThreadDepedencyTestApp.cpp

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <absl/strings/str_format.h>
+#include <absl/synchronization/mutex.h>
+
+#include <atomic>
+#include <chrono>
+#include <csignal>
+#include <iostream>
+#include <iterator>
+#include <mutex>
+#include <thread>
+
+#include "OrbitBase/ThreadUtils.h"
+
+static std::atomic<bool> exit_requested;
+static std::atomic<int> num;
+
+absl::Mutex test_mutex;
+
+constexpr int kNumOfThreads = 4;
+constexpr int kSigint = 2;
+
+std::thread my_threads[kNumOfThreads];
+
+void SignalHandler(int signum) {
+  std::cout << "Cleaning up!" << std::endl;
+  exit_requested.store(false);
+  num++;
+  for (auto& my_thread : my_threads) {
+    my_thread.join();
+  }
+  exit(signum);
+}
+
+static void DoWork(int thread_num) {
+  std::string thread_name = absl::StrFormat("Thread %d", thread_num);
+  orbit_base::SetCurrentThreadName(thread_name.c_str());
+  while (true) {
+    if (!exit_requested.load()) {
```

Comment:
You can reverse the meaning of true and false for `exit_requested`? Right now you're checking whether exit_requested becomes false to indicate that an exist was requested.